src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1194,6 +1268,28 @@ private int decodeIgnoreAll(
         return limit;
     }
 
+    private int decodeIgnorePlainFrame(
+        AmqpServer server,
+        long traceId,
+        long authorization,
+        long budgetId,
+        DirectBuffer buffer,
+        int offset,
+        int limit)
+    {
+        final int length = limit - offset;
+
+        int progress = offset;
+
+        if (length != 0)
+        {
+            progress = (int) (offset + server.decodableBodyBytes);
```

Comment:
This implementation relies on side effects of `decodePlainFrame` preceding it, but it would be better to make this `decodeIgnorePlainFrame` side effect free.

First, it needs to wrap the frame header and verify that the complete frame body is present, only then should it advance the progress to the frame body limit computed from the frame header fields alone.

This current implementation seems susceptible to an incomplete frame body.

Reply:
I'm setting `decoder = decodeIgnorePlainFrame` in `decodePerformative()`. So the `offset` in this `decodeIgnorePlainFrame()` starts from just after the frame header. So I cannot wrap the frame header here.

Reply:
Understood, then let's rename this as `decodeIgnoreFrameBody`.
The point about potentially advancing `progress` beyond `limit` still stands though, so you'll need to handle incrementally ignoring the remaining `server.decodableBytes` until it becomes `0` before updating the decoder to decode the next plain frame.